exporters/trace/jaeger/jaeger.go

```diff
@@ -214,14 +214,26 @@ func (e *Exporter) ExportSpans(ctx context.Context, ss []*export.SpanSnapshot) e
 		return nil
 	}
 
-	for _, span := range ss {
-		// TODO(jbd): Handle oversized bundlers.
-		err := e.bundler.Add(span, 1)
-		if err != nil {
-			return fmt.Errorf("failed to bundle %q: %w", span.Name, err)
+	var bundlerErr error
+	done := make(chan struct{})
+	go func() {
```

Comment:
Do we need to do this in a separate goroutine?

`AddWait` should return when the context is canceled or expired. This approach will introduce a possibility to return from the parent goroutine but it also introduces a context switch to see if the parent goroutine should return while the child is determining the same thing. This will introduce a minor goroutine leak while the child goroutine remains suspended. For high throughput systems with single threads this could create back pressure and possibly resource starvation.